File: DotnetBackend/Models/Room.cs
using System.ComponentModel.DataAnnotations;

namespace DotNetBackend.Models
{

    public class Room
    {
        [Key]
        public int ID { get; set; }

        [Required]
        [StringLength(64)]
        public string Name { get; set; }

        [Required]
        public double Lat { get; set; }

        [Required]
        public double Lon { get; set; }

        [Required]
        public double Alt { get; set; }

        [Required]
        public int Level { get; set; }

        [Required]
        [StringLength(128)]
        public string Site { get; set; } 
        
        public Room()
        {
            
        }
        
        public Room(RoomDTO roomDTO)
        {
            Name = roomDTO.Name;
            Lat = roomDTO.Lat;
            Lon = roomDTO.Lon;
            Alt = roomDTO.Alt;
            Level = roomDTO.Level;
            Site = roomDTO.Site;
        }
    }
}

File: FlutterApp/lib/Models/Room.dart

class Room {
  String name;
  double latitude;
  double longitude;
  double altitude;
  int level;
  String siteName;

  Room(
      {required this.name,
      required this.latitude,
      required this.longitude,
      required this.altitude,
      required this.level,
      required this.siteName});

  factory Room.fromJson(Map<String, dynamic> json) {
    return Room(
      name: json['name'] ?? '',
      latitude: (json['lat'] ?? 0.0).toDouble(),
      longitude: (json['lon'] ?? 0.0).toDouble(),
      altitude: (json['alt'] ?? 0.0).toDouble(),
      level: (json['level'] ?? 0).toInt(),
      siteName: json['siteName'] ?? '',
    );
  }
}
